admin-panel/cluster_analisys.py

```diff
@@ -0,0 +1,264 @@
+            threshold = 0.6  # нижняя граница для того, чтобы считать набор букв словом
+            s = ""
+
+            spec_words = [
+                "тюмгу",
+                "шкн",
+                "игип",
+                "фэи",
+                "соцгум",
+                "ипип",
+                "биофак",
+                "инзем",
+                "инхим",
+                "фти",
+                "инбио",
+                "ифк",
+                "ед",
+                "шпи",
+            ]  # временный костыль для нахождения абривиатур ТюмГУ
+            z = {i for i in punctuation}  # набор пунктуации
+            for word in words.split(" "):
+                # берём по слову, не нарушая пунктуации
+                x = word[-1]
+                if x in z:
+                    word = word[:-1]
+                else:
+                    x = ""
+                # проверяем слово, на абривиатуру ТюмГУ
+                if word.lower() in spec_words:
+                    score = 1
+                else:
+                    p = self.morph.parse(word)
+                    score = p[0].score
+                # добовляем слово в новую строку
+                if score >= threshold:
+                    s += " " + word + x
+            # возвращаем слово или False
+            if s == "":
+                return None
+            x1 = len(words.split(" ")) / 2
+            x2 = len(s.split(" ")) - 1
+            if x1 < x2:
+                return words
+            return None
+
+        df["text"] = df["text"].apply(del_spec_sim)
+        df["text"] = df["text"].apply(found_trash_words)
+        df = df.dropna()
+        return df
+
+    def vectorizing(self, df: pd.DataFrame) -> np.ndarray:
+        """Метод векторизации предложений, подлежащих анализу
+
+        Args:
+            df (pd.DataFrame): датафрейм содержания и даты появления вопроса
+
+        Returns:
+            np.ndarray: массив, содержащий в каждой строке векторные представления вопросов
+        """
+
+        vectorizer = TfidfVectorizer(min_df=2)
+        scaler = StandardScaler()
+
+        def lower_stopword_lemmatize(text: str) -> str:
+            """Возвращает леммы по предложениям
+
+            Args:
+                text (str): исходное предложение
+
+            Returns:
+                str: итоговые леммы
+            """
+
+            doc = self.nlp(str(text))
+            return " ".join(
+                [
+                    token.lemma_
+                    for token in doc
+                    if not token.is_stop and token.pos_ != "PUNCT"
+                ]
+            )
+
+        df["text_lemma"] = df["text"].apply(lower_stopword_lemmatize)
+
+        X = vectorizer.fit_transform(df["text_lemma"])
+        vectorizer.get_feature_names_out()
+
+        vectors = pd.DataFrame(
+            X.toarray().transpose(), vectorizer.get_feature_names_out()
+        )
+        scaler.fit(vectors)
+        vectors = scaler.transform(vectors).transpose()
```

Comment:
зачем преобразования в датафрейм и транспонирования?

Reply:
попробовать
```
from sklearn.pipeline import make_pipeline

vectorizer = make_pipeline(
    TfidfVectorizer(min_df=2),
    StandardScaler(with_mean=False)
)

return vectorizer.fit_transform(df["text_lemma"]) # если надо `.toarray()`
```
почему были нули? возможно, ответ есть в [доках](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html): 
> This scaler can also be applied to sparse CSR or CSC matrices by passing `with_mean=False` to avoid breaking the sparsity structure of the data.